RocketChat/Rocket.Chat
# fix: Routing fixed for private apps in marketplace section

## Proposed changes (including videos or screenshots)


This pull request fixes the irregular behavior of routing in the Private Apps section of the Marketplace. The issue was caused by the passing of three parameters (id of app, marketplaceVersion, and tab) to the route when opening an app's details. In the case of private apps, the marketplaceVersion was returning undefined, which caused the version to take on the value of the tab when redirecting. To resolve this issue, I have updated the code to pass the version of the app when the marketplaceVersion returns undefined. This fix ensures that the routes behave as expected in the Private Apps section of the Marketplace.

https://user-images.githubusercontent.com/89914602/225092278-48854c4e-9135-4f6a-ae3e-6f99250dfa45.mp4





## Issue(s)

#28378 
## Steps to test or reproduce

1. Open the Private apps section of marketplace. http://localhost:3000/marketplace/private/list
2. Click on any installed app , it will redirect to details page of the app.
3. Now open the logs of app and notice the route (/${version_of_app}/logs).
## Further comments
